### PR TITLE
EZP-28158: Test for EZP-28147 Updated user cannot log in

### PR DESCRIPTION
**Pull request by glye**
> Attempt at implementing https://jira.ez.no/browse/EZP-28158
> which is to add a test for https://jira.ez.no/browse/EZP-28147 aka. https://github.com/ezsystems/ezpublish-kernel/pull/2129
> Sent to QA

- [x] Test fails when reverting EZP-28147
- [x] Test passes when re-reverting EZP-28147

### CODE REVIEW

**Comment by glye** [comment]
Failing as it should on appveyor, this is promising :)

File: eZ/Publish/API/Repository/Tests/UserServiceTest.php [hunk 1]
```diff
@@ -1576,6 +1578,83 @@ public function testUpdateUser()
...
+
+        /* BEGIN: Use Case */
+        $user = $this->createUserVersion1();
+
+        $settingsProperty->setValue(
+            $userService,
+            [
+                'hashType' => User::PASSWORD_HASH_PHP_DEFAULT,
+            ] + $settingsProperty->getValue($userService)
+        );
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/UserServiceTest.php]: Note to self: Uh, we should consider exposing a possibility to update settings via API in the future.

Comment (glye) [reply eZ/Publish/API/Repository/Tests/UserServiceTest.php]: That would be useful here, yes :)

**Comment by glye** [comment]
Nitpick updates, thanks @alongosz

File: eZ/Publish/API/Repository/Tests/UserServiceTest.php [hunk 2]
```diff
@@ -1576,6 +1578,80 @@ public function testUpdateUser()
...
+        $userServiceProperty->setAccessible(true);
+        $userService = $userServiceProperty->getValue($signalSlotUserService);
+
+        $userServiceReflection = new ReflectionClass($userService);
+        $settingsProperty = $userServiceReflection->getProperty('settings');
+        $settingsProperty->setAccessible(true);
+        $settingsProperty->setValue(
+            $userService,
+            [
+                'hashType' => User::PASSWORD_HASH_MD5_USER,
```
Comment (andrerom) [inline eZ/Publish/API/Repository/Tests/UserServiceTest.php]: Can we move these constants to API? They kind of need to stay constant so could make sense there and would avoid adding more hard coding of Core logic in integration tests _(they are not supposed to known about Core internals)_. /cc @alongosz

**Comment by glye** [comment]
Update: Moved password hash constants to API User @andrerom @alongosz 
Also updated usage. Not strictly needed, but anyways...

**Comment by andrerom** [comment]
@alongosz Ok with the changes?

**Comment by glye** [comment]
Sent to QA. Should be fully BC but... 😈 🔍

**Pull request merged**